### PR TITLE
[AMORO-2359] Support log out of the ams dashboard

### LINKED ISSUE #2359: [Improvement]:  Support log out of the ams dashboard
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Supporting log out of the ams dashboard is preferred for a better user experience.

### How should we improve?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by tcodehuber**
## Why are the changes needed?

Close #2359.
Screenshot as follows:
screenshot-20231125-132743
screenshot-20231125-132805

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wangtaohz** [comment]
A very useful feature and thanks for you contribution! 👍 

cc @Aireed

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/LoginController.java [hunk 1]
```diff
@@ -56,6 +56,13 @@ public void login(Context ctx) {
     }
   }
 
+  /** handle logout post request. */
+  public void logout(Context ctx) {
+    ctx.consumeSessionAttribute("user");
```
Comment (shidayang) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/LoginController.java:61]: Why need this？

Comment (tcodehuber) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/LoginController.java:61]: Do you mean it(ctx.consumeSessionAttribute("user")) is redundant?

**Review by shidayang — APPROVED** [review]

**Comment by tcodehuber** [comment]
I have resolved conflicts, please review it again. @shidayang @wangtaohz

**Review by wangtaohz — APPROVED** [review]
LGTM.

I tested it locally and it worked well.
